dask_ml/model_selection/_incremental.py

```diff
@@ -958,6 +951,25 @@ def __init__(
         )
 
     def fit(self, X, y=None, **fit_params):
+        if "IncrementalSearchCV" in str(type(self)):
+            if self.decay_rate is None:
```

Comment:
Ideally users wouldn't need to use a filter to silence this. I think we can achieve it with something like

```
no_default = object()


class IncrementalSearchCV(...):
    def __init__(..., decay_rate=no_default, ...):
        ...
    def fit(self):
        if decay_rate is no_default: warnings.warn("deprecated... specify 'decay_rate=None' or use InverseDecaySearchCV to silence")
         # then set decay_rate=1.0 for backwards compat
        elif decay_rate is not None: warnings.warn("deprecated... use InverseDecaySearchCV")
```

That might obviate the need for `_check_decay_deprecation` since presumably `IncrementalSearchCV` would have a different default.

Then the three cases are

1. Not specified. Warn about using InverseDecaySearch if you want decay, o/w decay=None, set `decay_rate=1.0`, proceed
2. Specified with `decay_rate=None`: no warning
3. Specified with `decay_rate=number`: warn instructing to use InverseDecaySearchCV.

Reply:
I avoided this because I didn't want the user to change their code twice; once to set `decay_rate=None`, and once to remove `decay_rate` at some point in the future.

I've implemented your suggestion. Two deprecation cycles is clear.